rust/agama-cli/src/lib.rs

```diff
@@ -138,13 +153,63 @@ async fn build_manager<'a>() -> anyhow::Result<ManagerClient<'a>> {
     Ok(ManagerClient::new(conn).await?)
 }
 
+#[derive(PartialEq)]
+enum InsecureApi {
+    Secure,      // Remote api is secure
+    Insecure,    // Remote api is insecure - e.g. self-signed certificate
+    Forbidden, // Remote api is insecure and its use is forbidden (e.g. user decided not to use it)
+    Unreachable, // Remote api is unrecheable
+}
+
+/// Returns if insecure connection to remote api server is required and user allowed that
+async fn check_remote_api(api_url: String) -> Result<InsecureApi, ServiceError> {
+    // fake client used for remote site detection
+    let mut ping_client = BaseHTTPClient::default();
+    ping_client.base_url = api_url;
+
+    // decide whether access to remote site has to be insecure (self-signed certificate or not)
+    match ping_client.get::<HashMap<String, String>>("/ping").await {
+        Ok(res) => {
+            if res["status"] == "success" {
+                Ok(InsecureApi::Secure)
+            } else {
+                Ok(InsecureApi::Unreachable)
```

Comment:
Now that the enum is spelled correctly, I realize that it is unused. The caller does not care if it is `Secure` or `Unreachable`, so we might as well `Ok(_) => { Ok(InsecureApi::Secure }`, no?

Reply:
refactored a bit